### PR TITLE
feat(progress): ✨  add circular progress

### PR DESCRIPTION
**Pull request by navin-moorthy**
Circular Progress

### CODE REVIEW

File: src/circular-progress/CircularProgress.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+] = createContext<CircularProgressContext>({
+  name: "CircularProgressContext",
+  strict: false,
+});
+
+export { useCircularProgressContext };
+
+type CircularProgressRenderProps = {
+  children?:
+    | (({ ...state }: ProgressStateReturn) => JSX.Element)
```
Comment (anuraghazra) [inline src/circular-progress/CircularProgress.tsx:30]: `((state: ProgressStateReturn) => JSX.Element)`

File: src/circular-progress/CircularProgress.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  const context = React.useMemo(() => ({ state, size }), [state, size]);
+
+  React.useEffect(() => {
+    if (defaultValue !== undefined) setValue(defaultValue);
+  }, [defaultValue, setValue]);
+
+  return (
+    <CircularProgressProvider value={context}>
+      {children ? (
+        runIfFn(children, { ...state })
```
Comment (anuraghazra) [inline src/circular-progress/CircularProgress.tsx:69]: `runIfFn(children, state)`

File: src/circular-progress/CircularProgress.tsx (hunk 1 above)
Comment (navin-moorthy) [reply src/circular-progress/CircularProgress.tsx:30]: Why not spreading them ?
Whats the issue ?

Comment (navin-moorthy) [reply src/circular-progress/CircularProgress.tsx:30]: Got it !

File: src/circular-progress/CircularProgressBar.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        strokeWidth="10px"
+        strokeDashoffset="66"
+        strokeDasharray={strokeDasharray}
+        className={cx(
+          theme.circularProgress.bar.innerTrack.base,
+          isIndeterminate
+            ? theme.circularProgress.bar.innerTrack.indeterminate
+            : "",
+          innerTrackStyle,
+        )}
```
Comment (anuraghazra) [inline src/circular-progress/CircularProgressBar.tsx:63]: Extract all the inline cx() calls to upper level variable. looks cleaner.

Comment (anuraghazra) [reply src/circular-progress/CircularProgressBar.tsx:63]: No need to extract cx which has only two args. 

eg: `cx(theme.circularProgress.wrapper, className)`

Comment (navin-moorthy) [reply src/circular-progress/CircularProgressBar.tsx:63]: Right I sometimes forgot them 👍

**Pull request merged**